diesel/src/query_source/aliasing/aliased_field.rs

```diff
@@ -0,0 +1,108 @@
+use super::{Alias, AliasSource};
+
+use crate::dsl;
+use crate::expression::{
+    is_aggregate, AppearsOnTable, AsExpression, Expression, SelectableExpression, ValidGrouping,
+};
+use crate::expression_methods::{EqAll, ExpressionMethods};
+use crate::query_builder::{AstPass, FromClause, QueryFragment, QueryId, SelectStatement};
+use crate::sql_types;
+
+use crate::backend::Backend;
+use crate::query_source::{AppearsInFromClause, Column, Once, QuerySource};
+use crate::result::QueryResult;
+
+#[derive(Debug, Clone, Copy, Default)]
+/// Represents an aliased field (column) within diesel's query builder
+///
+/// See [alias!] for more details.
+pub struct AliasedField<S, F> {
+    pub(super) _alias_source: S,
+    pub(super) _field: F,
+}
+
+impl<S, C> QueryId for AliasedField<S, C>
+where
+    S: AliasSource + 'static,
+    S::Table: 'static,
+    C: Column<Table = S::Table> + 'static,
+{
+    type QueryId = Self;
+    const HAS_STATIC_QUERY_ID: bool = true;
```

Comment:
Again, this has only a static query id if the inner column has a static query id.

Reply:
Same as https://github.com/diesel-rs/diesel/pull/2254#discussion_r788717334